### DISCUSSION TITLE
Now dev 404 for dynamic routes

### DISCUSSION THREAD

**Discussion by Jmeza081** [vercel/vercel#4115] [General]
Hi there! Our organization has been using `Next.js` and `Now CLI` for a few of our projects. Over the last couple of months, we’ve come to discover a specific behavior that randomly occurs during development but is often difficult to remedy. 

## Description of the issue
When developing with the now CLI via the `now dev` command we occasionally begin to receive `404 File not found` errors on all dynamic routes within our `Next.js` applications.

A few things to notes about this issue:
* 	The issue with dynamic routes only occurs in development. Any staging or production environments deployed on `now` work perfectly fine.
*  Once the issue appears, it is difficult to remedy as we have tried multiple things to no avail.
* We have several projects configured in a similar fashion (See the `About our development environment` section) and all projects at random points begin to have display this issue as well.

### What we’ve tried with no luck
* Clearing all site-data, browser caches in the event that switching between multiple `Next.js` projects may be causing an issue
* Updating our `now` CLI to the latest version
	* We have also tried un-installing our `now`CLI and re-installing it
* We have cleared the `now` CLI cache located in the `library` folder
* We have tried updating our `Next.js` versions to the latest release

### About our development environment
* We have several `Next.js` projects which leverage a custom `now.json` config
* When developing our projects we use `now dev —local-config now.development.json —listen 3003` as our starting command
* We leverage the `File API` setup to configure the routing layer as opposed to the `routes: []` configuration in the `now.json` file

### A few leads we have discovered
Upon looking into the issue with dynamic routes, we decided to run our project using `--debug` flag. We quickly noticed that when the routing layer attempts to navigate to a dynamic route, the verbose output is the following before receiving the 404 error:
```
> [debug] [2020-04-21T20:50:20.300Z] GET /_next/static/development/pages/posts/edit/dfc42587-d5e5-43fd-91cf-4c3888aa2971.js
> [debug] [2020-04-21T20:50:20.302Z] Checking build result's 46 `routes` to match /_next/static/development/pages/posts/edit/dfc42587-d5e5-43fd-91cf-4c3888aa2971.js
> [debug] [2020-04-21T20:50:20.302Z] Found matching route http://localhost:61199/_next/static/development/pages/posts/edit/dfc42587-d5e5-43fd-91cf-4c3888aa2971.js for /_next/static/development/pages/posts/edit/dfc42587-d5e5-43fd-91cf-4c3888aa2971.js
> [debug] [2020-04-21T20:50:20.304Z] ProxyPass: http://localhost:61199/_next/static/development/pages/posts/edit/dfc42587-d5e5-43fd-91cf-4c3888aa2971.js
> [debug] [2020-04-21T20:50:20.334Z] GET /posts/edit/dfc42587-d5e5-43fd-91cf-4c3888aa2971
```
* The `ProxyPass` statement matches the route from `http://localhost:61199/` and would be expected to proxied to our `3003` port.
	* Accessing the dynamic route at `3003` directly results in the aforementioned `404 File not found` error
	* Accessing the dynamic route at `61199 ` works perfectly fine

We don’t know if this issue might be related to either an oversight on our end with regards to our `now` configuration or a potential bug with `now` that happens on localhost only.

Please let me know if there is any other information such as config files I can provide that may help us in exploring this issue further. I am curious if anyone has come across a similar issue like this before.

**Comment by coetry** [comment]
Hi there! What version of the CLI are you running? The latest version (18.0.0) should invoke your frameworks native dev script (`next dev` in this case). 

Can you try running `next dev` directly and see if you're seeing the same behavior?

**Reply by Jmeza081** [reply]
Thanks for the reply!

I upgraded to the latest now CLI version as you suggested and ran `next dev` as well as the regular `now dev`. Both resulted in the same error. 

I also upgraded my `Next` version to the latest release in the event it was causing an unforeseen backward compatibility issue. Both to no avail.

Is there any other configuration I might be able to provide to give you some insight? I will keep troubleshooting on my end in the meantime.
